### PR TITLE
PolicyReport spilt up per policy

### PR DESCRIPTION
**Pull request by prateekpandey14**
Signed-off-by: prateekpandey14 <prateek.pandey@nirmata.com>

### CODE REVIEW

**Review by chipzoller — APPROVED** [review]
In addition to one in-line comment, I'd like to see a plan for how the following scenarios will be addressed:

1. User updates a ClusterPolicy/Policy to add a rule which previously didn't exist.
2. User removes a rule from an existing ClusterPolicy/Policy.
3. User renames a ClusterPolicy/Policy.
4. How this will work with the new auto-gen capabilities (currently feature flagged) and if there are any modifications needed.

**Review by realshuting — COMMENTED** [review]
Added minor comments.

It would be good to address [these](https://github.com/kyverno/KDP/pull/24#pullrequestreview-1020075476) questions.

File: proposals/policy_report_spilt.md [hunk 1]
```diff
@@ -0,0 +1,156 @@
...
+```
+
+
+## Link to the Implementation PR: 
+
+- https://github.com/kyverno/kyverno/pull/4147
+
+# Migration (OPTIONAL)
+
+To enable this feature, user needs to enable the `splitPolicyReport: true` feature gate in kyverno container args, which is disable by default. Once enabled kyverno will delete the existing old reports and create the new reports on per policy bases in the background.
```
[the remark this answers is not in the document]
Comment (prateekpandey14) [reply proposals/policy_report_spilt.md:152]: Basically we want to delete the old report once the feature gate is set to true in kyverno container args (while container restart), do you see any other benefit with ConfigMap ?

**Comment by prateekpandey14** [comment]
>In addition to one in-line comment, I'd like to see a plan for how the following scenarios will be addressed:

Sure will add those details as well.

**Review by vyankyGH — APPROVED** [review]

**Pull request merged**